src/limitedmap.h

```diff
@@ -66,8 +66,7 @@ class limitedmap
     }
     void update(const_iterator itIn, const mapped_type& v)
     {
-        // TODO: When we switch to C++11, use map.erase(itIn, itIn) to get the non-const iterator.
-        iterator itTarget = map.find(itIn->first);
+        iterator itTarget = map.erase(itIn, itIn);
```

Comment:
This doesn't look right. Since erase returns the _next_ iterator, I'm not sure what good it would do here. It's also erasing without re-adding.

As-is, this appears to be modifying a random element.